tests/system/libraries/ChromeLib.py

```diff
@@ -103,76 +152,89 @@ def _writeTestFile(testCase) -> str:
 			f.write(fileContents)
 		return filePath
 
-	def _wasStartMarkerSpoken(self, speech: str):
-		if "document" not in speech:
-			return False
-		documentIndex = speech.index("document")
-		marker = ChromeLib._beforeMarker
-		return marker in speech and documentIndex < speech.index(marker)
-
-	def _waitForStartMarker(self, spy, lastSpeechIndex):
+	def _waitForStartMarker(self) -> bool:
 		""" Wait until the page loads and NVDA reads the start marker.
-		@param spy:
-		@type spy: SystemTestSpy.speechSpyGlobalPlugin.NVDASpyLib
-		@return: None
+		Depends on Chrome having focus, then tries to ensure that the document is focused and NVDA
+		virtual cursor is set to the "start marker"
+		@return: False on failure
 		"""
-		for i in range(3):  # set a limit on the number of tries.
-			builtIn.sleep("0.5 seconds")  # ensure application has time to receive input
-			spy.wait_for_speech_to_finish()
-			actualSpeech = spy.get_speech_at_index_until_now(lastSpeechIndex)
-			if self._wasStartMarkerSpoken(actualSpeech):
-				break
-			lastSpeechIndex = spy.get_last_speech_index()
-		else:  # Exceeded the number of tries
-			spy.dump_speech_to_log()
-			builtIn.fail(
-				"Unable to locate 'before sample' marker."
-				f" Too many attempts looking for '{ChromeLib._beforeMarker}'"
-				" See NVDA log for full speech."
-			)
+		spy = _NvdaLib.getSpyLib()
+		spy.emulateKeyPress('alt+d')  # focus the address bar, chrome shortcut
+		spy.wait_for_speech_to_finish()
+		addressSpeechIndex = spy.get_last_speech_index()
 
-	def _focusChrome(self, startsWithTestCaseTitle: re.Pattern):
-		""" Ensure chrome started and is focused.
-		Different versions of chrome have variations in how the title is presented.
-		This may mean that there is a separator between document name and application name.
-		E.G. "htmlTest   Google Chrome", "html – Google Chrome" or perhaps no application name at all.
-		Rather than try to get this right, just use the doc title.
-		If this continues to be unreliable we could use Selenium or similar to start chrome and inform us
-		when it is ready.
+		spy.emulateKeyPress('control+F6')  # focus web content, chrome shortcut.
+		spy.wait_for_speech_to_finish()
+		afterControlF6Speech = spy.get_speech_at_index_until_now(addressSpeechIndex)
+		if f"document\n{ChromeLib._beforeMarker}" not in afterControlF6Speech:
+			builtIn.log(afterControlF6Speech, level="DEBUG")
+			return False
+
+		spy.emulateKeyPress('control+home')  # ensure we start at the top of the document
+		controlHomeSpeechIndex = spy.get_last_speech_index()
+		spy.emulateKeyPress('numpad8')  # report current line
+		spy.wait_for_speech_to_finish()
+		afterNumPad8Speech = spy.get_speech_at_index_until_now(controlHomeSpeechIndex)
+		if ChromeLib._beforeMarker not in afterNumPad8Speech:
+			builtIn.log(afterNumPad8Speech, level="DEBUG")
+			return False
+		return True
+
+	def toggleFocusChrome(self):

```

Comment:
I don't think this function will work when running the tests locally, as users may have multiple windows open.
If something else "steals" focus from chrome, then `alt+tab` will focus the previously focused window, correct?

Reply:
Yes, this could be a problem. It's a different category of problem than the one I've seen locally: Chrome takes the foreground, has focus, but NVDA doesn't load a virtual buffer and stays silent, removing then restoring chrome as the foreground app resolves it.

I believe we have seen a situation on appveyor of some other window stealing focus after chrome has opened.
If this could happen at any moment, working around that would be very difficult in the system tests. Essentially you'd have to detect it and start the test again.